lib/js/src/rpc/messages/RegisterAppInterface.js

```diff
@@ -0,0 +1,381 @@
+
+    /**
+    * @return {String} the hash ID
+    */
+    getHashID() {
+        return this.getParameter(RegisterAppInterface.KEY_HASH_ID);
+    }
+
+    /**
+     * @param {DeviceInfo} deviceInfo
+     * @return {RegisterAppInterface}
+     */
+    setDeviceInfo(deviceInfo) {
+        this.validateType(DeviceInfo, deviceInfo);
+
+        this.setParameter(RegisterAppInterface.KEY_DEVICE_INFO, deviceInfo);
+        return this;
+    }
+
+    /**
+    * @return {DeviceInfo}
+    */
+    getDeviceInfo() {
+        return this.getParameter(RegisterAppInterface.KEY_DEVICE_INFO);
+    }
+
+
+    /**
+    * @param {String} appName This method should not be accessed directly by developers. Only set the full ID and this param will be set.
+    * @return {RegisterAppInterface}
+    */
+    _setAppId(appId) {
+        //TODO do we need this? this.validateType(String, appId);
+
+        this.setParameter(RegisterAppInterface.KEY_APP_ID, appId);
+        return this;
+    }
+
+    /**
+    * @return {String} the app id
+    */
+    getAppId() {
+        return this.getParameter(RegisterAppInterface.KEY_APP_ID);
+    }
+
+    /**
+    * @param {String} fullAppId 
+    * @return {RegisterAppInterface}
+    */
+    setFullAppId(fullAppId) {
+        //TODO do we need this? this.validateType(String, fullAppId);
+
+        if (fullAppID != null) {
+            fullAppID = fullAppID.toLowerCase();
+            setParameters(RegisterAppInterface.KEY_FULL_APP_ID, fullAppID);
+            let appID;
+            if (fullAppID.length() <= RegisterAppInterface.APP_ID_MAX_LENGTH) {
+                appID = fullAppID;
+            } else {
+                appID = fullAppID.replace("-", "").substring(0, RegisterAppInterface.APP_ID_MAX_LENGTH);
+            }
+            this._setAppId(appID);
+        } else {
+            setParameters(RegisterAppInterface.KEY_FULL_APP_ID, null);
+        }
+
+        return this;
+    }
+
+    /**
+    * @return {String} the app id
+    */
+    getFullAppId() {
+        return this.getParameter(RegisterAppInterface.KEY_FULL_APP_ID);
+    }
+
+
+    /**
+     * @param {AppInfo} appInfo
+     * @return {RegisterAppInterface}
+     */
+    setAppInfo(appInfo) {
+        this.validateType(AppInfo, appInfo);
+
+        this.setParameter(RegisterAppInterface.KEY_APP_INFO, appInfo);
+        return this;
+    }
+
+    /**
+    * @return {AppInfo}
+    */
+    getAppInfo() {
+        return this.getParameter(RegisterAppInterface.KEY_APP_INFO);
+    }
+
+    /**
+     * @param {TemplateColorScheme} dayColorScheme
+     * @return {RegisterAppInterface}
+     */
+    setDayColorScheme(dayColorScheme) {
+        this.validateType(TemplateColorScheme, dayColorScheme);
+
+        this.setParameter(RegisterAppInterface.KEY_DAY_COLOR_SCHEME, dayColorScheme);
+        return this;
+    }
+
+    /**
+    * @return {TemplateColorScheme}
+    */
+    getDayColorScheme() {
+        return this.getParameter(RegisterAppInterface.KEY_DAY_COLOR_SCHEME);
```

Comment:
As I understand there should be `this.getObject(TemplateColorScheme, RegisterAppInterface.KEY_DAY_COLOR_SCHEME);` instead of your `getParameter` is it right?

Reply:
You can directly comment code changes with the GitHub interface which makes it a lot easier to take the requests.